### PR TITLE
Add build config for TravisCI

### PR DESCRIPTION
**Pull request by DTMad**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+          - $HOME/.cache/go-build
+          - $HOME/gopath/pkg/mod
+
+  - &deploy
+    script: ./build/deploy.sh
+
+  - &create_manifest
+    stage: manifest
+    install: skip
+    git:
```
Comment (lrgar) [inline .travis.yml]: I guess this needs `language: minimal` or `generic` as well sinc Git clone is disabled: https://docs.travis-ci.com/user/languages/minimal-and-generic/

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      env: IMAGE=$OAO_IMAGE_DOCKER_HUB TAG="snapshot"
+      workspaces:
+        use:
+          - ws_amd
+          - dockerconfig
+
+    - <<: *deploy
+      name: Deploy snapshot quay AMD
+      arch: amd64
+      if: branch != master
```
Comment (lrgar) [inline .travis.yml]: I wonder if `branch != master` also includes release tags?

File: .travis.yml (hunk 1 above)
Comment (DTMad) [reply .travis.yml]: Changed to generic

File: .travis.yml (hunk 2 above)
Comment (DTMad) [reply .travis.yml]: Added an additional check if the tag is blank

**Pull request merged**